### ISSUE TITLE
[Android] Cannot add task 'uploadSourcemaps' as a task with that name already exists.

### ISSUE THREAD

**Issue by lukebars** [Instabug/Instabug-React-Native#989]
Android build is failing due to duplicate tasks with same name, something is wrong with flavor naming on version `11.12.0`. `11.10.0` works fine.

### Steps to Reproduce the Problem

`./gradlew clean bundleRelease`

### Expected Behavior

Build compiles successfully. 

### Actual Behavior

Byu

```console
FAILURE: Build failed with an exception.
* Where:
Script '/Users/App.Release/builds/aknPzrPd/0/our-app/node_modules/instabug-reactnative/android/sourcemaps.gradle' line: 23
* What went wrong:
Cannot add task 'uploadSourcemaps' as a task with that name already exists.
```

### SDK Version
```json
    "instabug-reactnative": "^11.12.0",
```

### React Native, iOS and Android Versions

```json
    "react-native": "^0.71.7",
```

**Comment by lukebars** [comment]
hi @a7medev, sure - here it goes:

```console
➜  android git:(cd437d0e) ✗ ./gradlew tasks
Configuration on demand is an incubating feature.

> Configure project :app
##############################

Using react.gradle in your project is deprecated! You should move to "apply plugin: com.facebook.react"
react.gradle is going to be removed in a future React Native project and your project will not build anymore.
You can use the template as a reference:
https://github.com/facebook/react-native/blob/main/template/android/app/build.gradle

##############################
WARNING:DSL element 'dexOptions' is obsolete and should be removed.
It will be removed in version 8.0 of the Android Gradle plugin.
Using it has no effect, and the AndroidGradle plugin optimizes dexing automatically.

> Configure project :react-native-blob-util
WARNING:Software Components will not be created automatically for Maven publishing from Android Gradle Plugin 8.0. To opt-in to the future behavior, set the Gradle property android.disableAutomaticComponentCreation=true in the `gradle.properties` file or use the new publishing DSL.

> Configure project :react-native-firebase_analytics
:react-native-firebase_analytics package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/analytics/package.json
:react-native-firebase_app package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/app/package.json
:react-native-firebase_analytics:firebase.bom using default value: 30.5.0
:react-native-firebase_analytics package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/analytics/package.json
:react-native-firebase_analytics:version set from package.json: 14.12.0 (14,12,0 - 14012000)
:react-native-firebase_analytics:android.compileSdk using custom value: 33
:react-native-firebase_analytics:android.targetSdk using custom value: 33
:react-native-firebase_analytics:android.minSdk using custom value: 23
:react-native-firebase_analytics:reactNativeAndroidDir /Users/lukebars/Desktop/product/x-app/node_modules/react-native/android

> Configure project :react-native-firebase_app
:react-native-firebase_app package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/app/package.json
:react-native-firebase_app:firebase.bom using default value: 30.5.0
:react-native-firebase_app:play.play-services-auth using default value: 20.3.0
:react-native-firebase_app package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/app/package.json
:react-native-firebase_app:version set from package.json: 14.12.0 (14,12,0 - 14012000)
:react-native-firebase_app:android.compileSdk using custom value: 33
:react-native-firebase_app:android.targetSdk using custom value: 33
:react-native-firebase_app:android.minSdk using custom value: 23
:react-native-firebase_app:reactNativeAndroidDir /Users/lukebars/Desktop/product/x-app/node_modules/react-native/android

> Configure project :react-native-firebase_auth
:react-native-firebase_auth package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/auth/package.json
:react-native-firebase_app package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/app/package.json
:react-native-firebase_auth:firebase.bom using default value: 30.5.0
:react-native-firebase_auth package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/auth/package.json
:react-native-firebase_auth:version set from package.json: 14.12.0 (14,12,0 - 14012000)
:react-native-firebase_auth:android.compileSdk using custom value: 33
:react-native-firebase_auth:android.targetSdk using custom value: 33
:react-native-firebase_auth:android.minSdk using custom value: 23
:react-native-firebase_auth:reactNativeAndroidDir /Users/lukebars/Desktop/product/x-app/node_modules/react-native/android

> Configure project :react-native-firebase_crashlytics
:react-native-firebase_crashlytics package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/crashlytics/package.json
:react-native-firebase_app package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/app/package.json
:react-native-firebase_crashlytics:firebase.bom using default value: 30.5.0
:react-native-firebase_crashlytics package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/crashlytics/package.json
:react-native-firebase_crashlytics:version set from package.json: 14.12.0 (14,12,0 - 14012000)
:react-native-firebase_crashlytics:android.compileSdk using custom value: 33
:react-native-firebase_crashlytics:android.targetSdk using custom value: 33
:react-native-firebase_crashlytics:android.minSdk using custom value: 23
:react-native-firebase_crashlytics:reactNativeAndroidDir /Users/lukebars/Desktop/product/x-app/node_modules/react-native/android

> Configure project :react-native-firebase_database
:react-native-firebase_database package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/database/package.json
:react-native-firebase_app package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/app/package.json
:react-native-firebase_database:firebase.bom using default value: 30.5.0
:react-native-firebase_database package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/database/package.json
:react-native-firebase_database:version set from package.json: 14.12.0 (14,12,0 - 14012000)
:react-native-firebase_database:android.compileSdk using custom value: 33
:react-native-firebase_database:android.targetSdk using custom value: 33
:react-native-firebase_database:android.minSdk using custom value: 23
:react-native-firebase_database:reactNativeAndroidDir /Users/lukebars/Desktop/product/x-app/node_modules/react-native/android

> Configure project :react-native-firebase_remote-config
:react-native-firebase_remote-config package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/remote-config/package.json
:react-native-firebase_app package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/app/package.json
:react-native-firebase_remote-config:firebase.bom using default value: 30.5.0
:react-native-firebase_remote-config package.json found at /Users/lukebars/Desktop/product/x-app/node_modules/@react-native-firebase/remote-config/package.json
:react-native-firebase_remote-config:version set from package.json: 14.12.0 (14,12,0 - 14012000)
:react-native-firebase_remote-config:android.compileSdk using custom value: 33
:react-native-firebase_remote-config:android.targetSdk using custom value: 33
:react-native-firebase_remote-config:android.minSdk using custom value: 23
:react-native-firebase_remote-config:reactNativeAndroidDir /Users/lukebars/Desktop/product/x-app/node_modules/react-native/android

> Configure project :react-native-flipper
WARNING:Software Components will not be created automatically for Maven publishing from Android Gradle Plugin 8.0. To opt-in to the future behavior, set the Gradle property android.disableAutomaticComponentCreation=true in the `gradle.properties` file or use the new publishing DSL.

> Configure project :react-native-reanimated
AAR for react-native-reanimated has been found
/Users/lukebars/Desktop/product/x-app/node_modules/react-native-reanimated/android/react-native-reanimated-71-hermes.aar

FAILURE: Build failed with an exception.

* Where:
Script '/Users/lukebars/Desktop/product/x-app/node_modules/instabug-reactnative/android/sourcemaps.gradle' line: 23

* What went wrong:
Cannot add task 'uploadSourcemaps' as a task with that name already exists.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.5.1/userguide/command_line_interface.html#sec:command_line_warnings
```

**Comment by lukebars** [comment]
@a7medev 

<details>
<summary>./gradlew tasks output without instabug</summary>

```console
------------------------------------------------------------
Tasks runnable from root project 'Greatness'
------------------------------------------------------------

Android tasks
-------------
androidDependencies - Displays the Android dependencies of the project.
signingReport - Displays the signing info for the base and test modules
sourceSets - Prints out all the source sets defined in this project.

Build tasks
-----------
assemble - Assemble main outputs for all the variants.
assembleAmazon - Assembles main outputs for all Amazon variants.
assembleAndroidTest - Assembles all the Test applications.
assembleDebug - Assembles main outputs for all Debug variants.
assemblePlay - Assembles main outputs for all Play variants.
assembleRelease - Assembles main outputs for all Release variants.
build - Assembles and tests this project.
buildDependents - Assembles and tests this project and all projects that depend on it.
buildKotlinToolingMetadata - Build metadata json file containing information about the used Kotlin tooling
buildNeeded - Assembles and tests this project and all projects it depends on.
bundle - Assemble bundles for all the variants.
clean - Deletes the build directory.
compileAmazonDebugAndroidTestSources
compileAmazonDebugSources
compileAmazonDebugUnitTestSources
compileAmazonReleaseSources
compileAmazonReleaseUnitTestSources
compileDebugAndroidTestSources
compileDebugSources
compileDebugUnitTestSources
compilePlayDebugAndroidTestSources
compilePlayDebugSources
compilePlayDebugUnitTestSources
compilePlayReleaseSources
compilePlayReleaseUnitTestSources
compileReleaseSources
compileReleaseUnitTestSources
extractAmazonDebugAnnotations - Extracts Android annotations for the amazonDebug variant into the archive file
extractAmazonReleaseAnnotations - Extracts Android annotations for the amazonRelease variant into the archive file
extractDebugAnnotations - Extracts Android annotations for the debug variant into the archive file
extractPlayDebugAnnotations - Extracts Android annotations for the playDebug variant into the archive file
extractPlayReleaseAnnotations - Extracts Android annotations for the playRelease variant into the archive file
extractReleaseAnnotations - Extracts Android annotations for the release variant into the archive file

Build Setup tasks
-----------------
init - Initializes a new Gradle build.
wrapper - Generates Gradle wrapper files.

Firebase Crashlytics tasks
--------------------------
generateCrashlyticsSymbolFileRelease - Generate native symbol file used by Crashlytics to symbolicate native (NDK) stack traces
injectCrashlyticsMappingFileIdDebug - Injects a mapping file id into the app, used by Crashlytics for deobfuscation.
injectCrashlyticsMappingFileIdRelease - Injects a mapping file id into the app, used by Crashlytics for deobfuscation.
uploadCrashlyticsSymbolFileRelease - Upload native (NDK) symbol files to Crashlytics to symbolicate stack traces

Help tasks
----------
buildEnvironment - Displays all buildscript dependencies declared in root project 'Greatness'.
dependencies - Displays all dependencies declared in root project 'Greatness'.
dependencyInsight - Displays the insight into a specific dependency in root project 'Greatness'.
help - Displays a help message.
javaToolchains - Displays the detected java toolchains.
outgoingVariants - Displays the outgoing variants of root project 'Greatness'.
projects - Displays the sub-projects of root project 'Greatness'.
properties - Displays the properties of root project 'Greatness'.
resolvableConfigurations - Displays the configurations that can be resolved in root project 'Greatness'.
tasks - Displays the tasks runnable from root project 'Greatness' (some of the displayed tasks may belong to subprojects).

Install tasks
-------------
installAmazonDebugAndroidTest - Installs the android (on device) tests for the AmazonDebug build.
installDebug - Installs the Debug build.
installDebugAndroidTest - Installs the android (on device) tests for the Debug build.
installPlayDebugAndroidTest - Installs the android (on device) tests for the PlayDebug build.
installRelease - Installs the Release build.
uninstallAll - Uninstall all applications.
uninstallAmazonDebugAndroidTest - Uninstalls the android (on device) tests for the AmazonDebug build.
uninstallDebug - Uninstalls the Debug build.
uninstallDebugAndroidTest - Uninstalls the android (on device) tests for the Debug build.
uninstallPlayDebugAndroidTest - Uninstalls the android (on device) tests for the PlayDebug build.
uninstallRelease - Uninstalls the Release build.

Publishing tasks
----------------
publish - Publishes all publications produced by this project.
publishToMavenLocal - Publishes all Maven publications produced by this project to the local Maven cache.

React tasks
-----------
bundleDebugJsAndAssets - bundle JS and assets for Debug.
bundleReleaseJsAndAssets - bundle JS and assets for Release.
copyDebugBundledJs - copy bundled JS into Debug.
copyReleaseBundledJs - copy bundled JS into Release.
createBundleReleaseJsAndAssets

Verification tasks
------------------
check - Runs all checks.
checkJetifier - Checks whether Jetifier is needed for the current project
connectedAmazonDebugAndroidTest - Installs and runs the tests for amazonDebug on connected devices.
connectedAndroidTest - Installs and runs instrumentation tests for all flavors on connected devices.
connectedCheck - Runs all device checks on currently connected devices.
connectedDebugAndroidTest - Installs and runs the tests for debug on connected devices.
connectedPlayDebugAndroidTest - Installs and runs the tests for playDebug on connected devices.
deviceAndroidTest - Installs and runs instrumentation tests using all Device Providers.
deviceCheck - Runs all device checks using Device Providers and Test Servers.
lint - Runs lint on the default variant.
lintAmazonDebug - Print text output from the corresponding lint report task
lintAmazonRelease - Print text output from the corresponding lint report task
lintAnalyzeAmazonDebug - Run lint analysis on the amazonDebug variant
lintAnalyzeAmazonRelease - Run lint analysis on the amazonRelease variant
lintAnalyzeDebug - Run lint analysis on the debug variant
lintAnalyzePlayDebug - Run lint analysis on the playDebug variant
lintAnalyzePlayRelease - Run lint analysis on the playRelease variant
lintAnalyzeRelease - Run lint analysis on the release variant
lintDebug - Print text output from the corresponding lint report task
lintFix - Runs lint on the default variant and applies any safe suggestions to the source code.
lintFixAmazonDebug - Fix lint on the amazonDebug variant
lintFixAmazonRelease - Fix lint on the amazonRelease variant
lintFixDebug - Fix lint on the debug variant
lintFixPlayDebug - Fix lint on the playDebug variant
lintFixPlayRelease - Fix lint on the playRelease variant
lintFixRelease - Fix lint on the release variant
lintPlayDebug - Print text output from the corresponding lint report task
lintPlayRelease - Print text output from the corresponding lint report task
lintRelease - Print text output from the corresponding lint report task
lintReportAmazonDebug - Run lint on the amazonDebug variant
lintReportAmazonRelease - Run lint on the amazonRelease variant
lintReportDebug - Run lint on the debug variant
lintReportPlayDebug - Run lint on the playDebug variant
lintReportPlayRelease - Run lint on the playRelease variant
lintReportRelease - Run lint on the release variant
lintVitalAnalyzeRelease - Run lint analysis with only the fatal issues enabled on the release variant
lintVitalRelease - Print text output from the corresponding lint report task
lintVitalReportRelease - Run lint with only the fatal issues enabled on the release variant
test - Run unit tests for all variants.
testAmazonDebugUnitTest - Run unit tests for the amazonDebug build.
testAmazonReleaseUnitTest - Run unit tests for the amazonRelease build.
testDebugUnitTest - Run unit tests for the debug build.
testPlayDebugUnitTest - Run unit tests for the playDebug build.
testPlayReleaseUnitTest - Run unit tests for the playRelease build.
testReleaseUnitTest - Run unit tests for the release build.
updateLintBaseline - Updates the lint baseline using the default variant.
updateLintBaselineAmazonDebug - Update the lint baseline using the amazonDebug variant
updateLintBaselineAmazonRelease - Update the lint baseline using the amazonRelease variant
updateLintBaselineDebug - Update the lint baseline using the debug variant
updateLintBaselinePlayDebug - Update the lint baseline using the playDebug variant
updateLintBaselinePlayRelease - Update the lint baseline using the playRelease variant
updateLintBaselineRelease - Update the lint baseline using the release variant
```

</details>

**Comment by a7medev** [comment]
@lukebars I see both `createBundleReleaseJsAndAssets` and `bundleReleaseJsAndAssets` are present in the tasks while you should have only one of them (`createBundleReleaseJsAndAssets` in this case). That's what's causing the `uploadSourcemaps` task to be created twice.
Can you please share the version of `react-native-gradle-plugin` used in the app? You can get it using the following command:

```sh
npm list react-native-gradle-plugin
```

**Comment by lukebars** [comment]
```console
yarn list v1.22.19
└─ react-native-gradle-plugin@0.71.18
```

**Comment by a7medev** [comment]
Hi @lukebars! We have made a fix for this issue in #991 and created a snapshot for you to test the changes, which you can install using the command below:

```sh
npm install https://github.com/Instabug/Instabug-React-Native\#snapshot/11.12.0-MOB-12473
```

Please let us know if that fixes your issue! 🙏🏼

**Comment by lukebars** [comment]
@a7medev looks good! Thanks!
